lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/HdfsFileIterator.java

```diff
@@ -68,11 +68,14 @@ public FileEntry next()
         }
 
         String root = listingPath.toUri().getPath();
+        if (!root.endsWith("/")) {
+            root += "/";
+        }
         String path = status.getPath().toUri().getPath();
 
         verify(path.startsWith(root), "iterator path [%s] not a child of listing path [%s] for location [%s]", path, root, listingLocation);
 
-        Location location = listingLocation.appendPath(path.substring(root.length() + 1));
+        Location location = listingLocation.appendPath(path.substring(root.length()));
```

Comment:
I don't understand this logic: we're stripping `root` prefix (which comes from `listingPath`) to prepend `listingPath`. does it yield `path` back? or something else?
let's have explanatory comment